exonum/src/events/noise/tests.rs

```diff
@@ -84,3 +99,253 @@ fn test_converted_keys_handshake() {
     h_r.into_transport_mode()
         .expect("Unable to transition session into transport mode");
 }
+
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub enum HandshakeStep {
+    None,
+    EphemeralKeyExchange(u8, usize),
+    StaticKeyExchange(u8, usize),
+}
+
+const MAX_MESSAGE_LEN: usize = 128;
+
+const EMPTY_MESSAGE: usize = 0;
+const STANDARD_MESSAGE: usize = MAX_MESSAGE_LEN;
+const LARGE_MESSAGE: usize = NOISE_MAX_MESSAGE_LENGTH + 1;
+
+impl HandshakeParams {
+    fn default_test_params() -> Self {
+        let (public_key, secret_key) = gen_keypair_from_seed(&Seed::new([0; 32]));
+        HandshakeParams {
+            max_message_len: MAX_MESSAGE_LEN as u32,
+            public_key,
+            secret_key,
+        }
+    }
+}
+
+#[test]
+#[should_panic(expected = "WrongMessageLength(0)")]
+fn test_noise_handshake_errors_ee_empty() {
+    let addr: SocketAddr = "127.0.0.1:45003".parse().unwrap();
+    let step = HandshakeStep::EphemeralKeyExchange(1, EMPTY_MESSAGE);
+    let (sender_err, listener_err) = wait_for_handshake_result(&addr, step);
+
+    assert!(sender_err.is_err());
+    listener_err.unwrap()
```

Comment:
Maybe use [`assert_matches!(..)`](https://docs.rs/assert_matches/1.2.0/assert_matches/macro.assert_matches.html) here (and below)? 
It's already used in testkit, so I guess bringing that dev-dependency isn't a big evil.